terraform/nix-build/main.tf

```
@@ -1,8 +1,12 @@
+locals {
+nix_options = "{\"options\": {${join(", ", [for k, v in var.nix_options :  "\"${k}\": \"${v}\""])} } }"
+}
```

Terraform has actually a nice function for this:

```suggestion
locals {
  nix_options = jsonencode({
    options = { for k, v in var.nix_options : k => v }
  })
}
```